static/js/common/fxa-login.js

```diff
@@ -5,8 +5,13 @@
         oauthHost: config.oauthHost,
     });
 
-    function currentPath() {
-        return location.pathname + location.search + location.hash;
+    function nextPath() {
+        var to = new Uri(location).getQueryParamValue('to');
```

Comment:
The `new` happens before the member access, I guess because they have the same [precedence](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Operator_Precedence#Table). I could add parens but they don't seem to be needed in JS unlike Java.